lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/FileBasedSystemAccessControlModule.java

```diff
@@ -0,0 +1,151 @@
+import static io.airlift.configuration.ConfigBinder.configBinder;
+import static io.airlift.http.client.HttpClientBinder.httpClientBinder;
+import static io.trino.plugin.base.security.CatalogAccessControlRule.AccessMode.ALL;
+import static io.trino.plugin.base.security.FileBasedAccessControlConfig.SECURITY_REFRESH_PERIOD;
+import static io.trino.spi.StandardErrorCode.CONFIGURATION_INVALID;
+import static java.lang.String.format;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+
+public class FileBasedSystemAccessControlModule
+        extends AbstractConfigurationAwareModule
+{
+    private static final Logger log = Logger.get(FileBasedSystemAccessControlModule.class);
+    private static final String HTTP_CLIENT_NAME = "access-control";
+
+    public FileBasedSystemAccessControlModule(Map<String, String> config)
+    {
+        super();
+        this.setConfigurationFactory(new ConfigurationFactory(config));
+    }
+
+    @Override
+    public void setup(Binder binder)
+    {
+        FileBasedAccessControlConfig configuration = buildConfigObject(FileBasedAccessControlConfig.class);
+        if (configuration.isRest()) {
+            binder.bind(new TypeLiteral<Supplier<FileBasedSystemAccessControlRules>>() {})
+                    .to(RestFileBasedSystemAccessControlRulesProvider.class)
+                    .in(Scopes.SINGLETON);
+            httpClientBinder(binder).bindHttpClient(HTTP_CLIENT_NAME, ForAccessControlRules.class)
+                    .withConfigDefaults(config -> config
+                            .setRequestTimeout(Duration.succinctDuration(10, TimeUnit.SECONDS))
+                            .setSelectorCount(1)
+                            .setMinThreads(1));
+            configBinder(binder).bindConfig(HttpClientConfig.class, HTTP_CLIENT_NAME);
+        }
+        else {
+            binder.bind(new TypeLiteral<Supplier<FileBasedSystemAccessControlRules>>() {})
+                    .toProvider(() -> new LocalFileAccessControlRulesProvider<>(configuration, FileBasedSystemAccessControlRules.class))
+                    .in(Scopes.SINGLETON);
+        }
+        configBinder(binder).bindConfig(FileBasedAccessControlConfig.class);
+    }
+
+    @Inject
+    @Provides
+    public SystemAccessControl getSystemAccessControl(FileBasedAccessControlConfig config,
+            Supplier<FileBasedSystemAccessControlRules> rulesProvider)
+    {
+        String configFilePath = config.getConfigFilePath();
+
+        if (config.getRefreshPeriod() != null) {
+            Duration refreshPeriod;
+            try {
+                refreshPeriod = config.getRefreshPeriod();
+            }
+            catch (IllegalArgumentException e) {
+                throw invalidRefreshPeriodException(config, configFilePath);
+            }
+            if (refreshPeriod.toMillis() == 0) {
+                throw invalidRefreshPeriodException(config, configFilePath);
+            }
```

Comment:
Can we move this validation part of the config like `public Optional<@MinDuration("1ms") Duration>`.

Reply:
We can use `Optional` too. Maybe it could go as a preparatory commit. WDYT ?

Reply:
refactored - actually it was dead code - refreshperiod validation are based on annotations and included in config class